src/glang/driver.py

```diff
@@ -31,15 +31,32 @@ def run_checked(args: list[str | Path], stdin: bytes | None = None) -> bytes:
         capture_output=True,
         check=False,
     )
-    if result.returncode != 0:
+    if result.stderr:
         command_str = " ".join(str(arg) for arg in args)
```

Comment:
You didn't touch this lines, but this looks like a job for `shlex.join` instead of `str.join`